lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/mapping/DefaultIdentifierMapping.java

```diff
@@ -11,26 +11,17 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.trino.plugin.jdbc.mapping;
+package io.trino.plugin.base.mapping;
 
-import com.google.inject.Inject;
 import io.trino.spi.security.ConnectorIdentity;
 
-import java.sql.Connection;
-
 import static java.util.Locale.ENGLISH;
-import static java.util.Objects.requireNonNull;
 
 public class DefaultIdentifierMapping
         implements IdentifierMapping
 {
-    private final RemoteIdentifierSupplier remoteIdentifierSupplier;
-
-    @Inject
-    public DefaultIdentifierMapping(RemoteIdentifierSupplier remoteIdentifierSupplier)
-    {
-        this.remoteIdentifierSupplier = requireNonNull(remoteIdentifierSupplier, "remoteIdentifierSupplier is null");
-    }
+    // Caching this on a field is LazyConnectorFactory friendly
+    private Boolean storesUpperCaseIdentifiers;
```

Comment:
I think this "caching" should be part of specific `RemoteIdentifiers` where it is expensive to return that infromation.

Reply:
we could remove this caching from `DefaultIdentifierMapping`
and add `CachingCaseSensitivityItIdentifierMapping`

But what to do with bindings? because now `DefaultIdentifierMapping` is binded as Default one should we instead bind as Default -  `CachingCaseSensitivityItIdentifierMapping` or introduce additional config property ?

Reply:
No. I don't mean `IdentifierMapping`, but `RemoteIdentifiers`. Notice that Druid `RemoteIdentifiers` is simply returning `false` so no caching is needed in such case, in default jdbc `RemoteIdentifiers` caching would be helpful.

Reply:
Moved caching to RemoteIdentifiers